applications/tari_console_wallet/src/main.rs

```diff
@@ -111,30 +124,188 @@ fn main_inner() -> Result<(), ExitCodes> {
-        runtime.handle().enter(|| ui::run(app))?;
-        println!("The wallet is shutting down.");
-        info!(
-            target: LOG_TARGET,
-            "Termination signal received from user. Shutting wallet down."
-        );
-        // TODO: Shutdown wallet
-        Ok(())
-    } else {
-        println!("Starting grpc server");
-        runtime
-            .block_on(run_grpc(grpc, node_config.grpc_wallet_address))
-            .map_err(ExitCodes::GrpcError)?;
-        println!("Shutting down");
-        Ok(())
+    let wallet_mode = match (bootstrap.daemon_mode, bootstrap.input_file) {
+        (false, None) => WalletMode::Tui,
+        (true, None) => WalletMode::Grpc,
+        (false, Some(path)) => WalletMode::Script(path),
+        (true, Some(_)) => WalletMode::Invalid,
+    };
+
+    match wallet_mode {
+        WalletMode::Tui => tui_mode(runtime, grpc, node_config, node_identity, wallet),
+        WalletMode::Grpc => grpc_mode(runtime, grpc, node_config),
+        WalletMode::Script(path) => script_mode(runtime, path, wallet),
+        WalletMode::Invalid => Err(ExitCodes::InputError(
+            "Can't run GRPC and file input mode simultaneously!".to_string(),
+        )),
+    }
+}
+
+fn script_mode(mut runtime: Runtime, path: PathBuf, wallet: WalletThreadSafe) -> Result<(), ExitCodes> {
+    // todo: println should be logs
+    let script = fs::read_to_string(path).map_err(|e| ExitCodes::InputError(e.to_string()))?;
+
+    if script.is_empty() {
+        return Err(ExitCodes::InputError("Input file is empty!".to_string()));
+    };
+
+    let mut commands = Vec::new();
+
+    for command in script.lines() {
+        // allow "comments" in the script file to start with '#'
+        if !command.starts_with('#') {
+            // parse the command
+            commands.push(parse_command(command)?);
+        }
+    }
+
+    println!("Starting command runner");
+    runtime.block_on(command_runner(commands, wallet))?;
+    println!("Shutting down");
+
+    Ok(())
+}
+
+async fn command_runner(commands: Vec<ParsedCommand>, wallet: WalletThreadSafe) -> Result<(), ExitCodes> {
+    // todo: consolidate "fee per gram" in codebase
+    let fee_per_gram = 25 * uT;
+    let mut wallet_transaction_service = wallet.read().await.transaction_service.clone();
+    let mut output_service = wallet.read().await.output_manager_service.clone();
+    let mut event_stream = wallet_transaction_service.get_event_stream_fused();
+
+    // track tx ids
+    let mut tx_ids = Vec::new();
+
+    for parsed in commands {
+        println!("{:?}", parsed);
+
+        use ParsedArgument::*;
+        match parsed.command {
+            BaseNodeCommand::GetBalance => match output_service.get_balance().await {
+                Ok(balance) => println!("Balance: {:?}", balance),
+                Err(e) => println!("output manager error! {:?}", e),
+            },
+            // todo: refactor to function
+            BaseNodeCommand::SendTari => {
```

Comment:
There should probably be two enums: BaseNodeCommand, WalletCommand

Reply:
Do you suggest a completely separate enum or some way of inheriting or reusing

Reply:
I think completely separate. The base node will eventually not accept WalletCommand commands